File: algorithms/0219.contains-duplicate-ii/solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn contains_nearby_duplicate(nums: Vec<i32>, k: i32) -> bool {
        let mut map = HashMap::new();

        for i in 0..nums.len() {
            match map.insert(nums[i], i) {
                Some(j) => {
                    if (i - j) as i32 <= k {
                        return true;
                    }
                }
                _ => continue,
            }
        }

        false
    }
}

File: algorithms/0219.contains-duplicate-ii/solution.go
func containsNearbyDuplicate(nums []int, k int) bool {
	m := make(map[int]int, len(nums))

	for i, num := range nums {
		if j, ok := m[num]; ok {
			if i-j <= k {
				return true
			}
		}
		m[num] = i
	}

	return false
}
